### ISSUE TITLE
Prediction samples/docs

### ISSUE THREAD

**Issue by mikeun** [ppwwyyxx/tensorpack#434]
Usage Questions:

1. can you provide clear docs about prediction stage? how can I write new predictor that uses model that I aleady have?
2. can you show examples of usage already learned model with new data?

**Comment by ppwwyyxx** [comment]
After training you can use `OfflinePredictor` with `PredictConfig` to get a callable. It is quite easy -- you usually only need to provide 4 arguments (model, where to load, input, output). Docs is [here](http://tensorpack.readthedocs.io/en/latest/modules/predict.html#tensorpack.predict.PredictConfig). Most examples contain the use of `OfflinePredictor` if you could grep.

We won't provide more sophisticated use -- because it's just a normal tensorflow model and you can use whatever features of tensorflow for inference.

**Comment by mikeun** [comment]
Thank you. 
I see that it is just tensorflow graph, but your queue to plug new data into it - it is not just "feed" ).
I use your tensorpack and it works well to push my GPU to 100% - 
I added stat function like confusion matrix.
'''
[32m[1007 15:52:17 @monitor.py:355][0m confusion matrix of class 0: | 38643|  2818|   431|    86|, Recall:92.06, Precision: 84.61
[32m[1007 15:52:17 @monitor.py:355][0m confusion matrix of class 1: |  6174| 12156|  1510|    59|, Recall:61.09, Precision: 72.79
[32m[1007 15:52:17 @monitor.py:355][0m confusion matrix of class 2: |   795|  1662|  6974|   313|, Recall:71.57, Precision: 76.09
[32m[1007 15:52:17 @monitor.py:355][0m confusion matrix of class 3: |    58|    63|   250|  1736|, Recall:82.39, Precision: 79.12
'''
but it adds new line with new epoch in the tensorboard scalars. 
![confusion_matrix_tensorboard](https://user-images.githubusercontent.com/28757897/31314917-cdc1cd08-ac14-11e7-8950-1f6c0f2814a3.png)

Question:
1) how can I turn off output my confusion matrix to tensorboard
2) why my test step (inference with test data) is not the same (quantities of data-points in the confusion matrix inside one class are not the same)?
'''
[0m confusion matrix of class 0: | 39992|  2136|   128|   104|
[0m confusion matrix of class 1: |  6744| 11696|   704|    80|
[0m confusion matrix of class 2: |   896|  1568|  7160|   344|
[0m confusion matrix of class 3: |    48|    40|   192|  1896|
'''
compare with the same test set but another run:
'''
[0m confusion matrix of class 0: | 38643|  2818|   431|    86|
[0m confusion matrix of class 1: |  6174| 12156|  1510|    59|
[0m confusion matrix of class 2: |   795|  1662|  6974|   313|
[0m confusion matrix of class 3: |    58|    63|   250|  1736|
'''

**Comment by ppwwyyxx** [comment]
1. Just print the matrix in your code and don't use `monitors.put`. Monitors don't support "matrix" anyway.
2. Your inference dataflow might not produce the same set of data every time, e.g. due to batching, random shuffling.

**Comment by mikeun** [comment]
2. My DataFlow = DataPoints are dumped to tfrecords by your dump_dataflow_to_tfrecord() to make  data injection faster (I added compress option to make result files much smaller). So test set is the same all times:
'''
test = TFRecordData('./test_.tfrecords', size=73768)
test = PrefetchDataZMQ(test, nr_proc=4, hwm=100)
test = BatchData(test, BATCH_SIZE)
'''
maybe that is consequent of "nr_proc=4"?

**Comment by ppwwyyxx** [comment]
prefetch with nr_proc>1 will not give you the same data. See `http://tensorpack.readthedocs.io/en/latest/modules/dataflow.html#tensorpack.dataflow.PrefetchDataZMQ`.

When `73768 % BATCH_SIZE != 0`, use of batch can also be a problem (missing a few datapoints in the end). You can use `remainder=True` (http://tensorpack.readthedocs.io/en/latest/modules/dataflow.html#tensorpack.dataflow.BatchData)
